pkmodel/protocol.py

```diff
@@ -36,18 +36,20 @@ def __init__(self):
 
     def read_config(self, file_dir):
         #get current directory and add to file_dir
-        config_file = open("file_dir", "r")
-        dictionaries_list_str = config_file  # .split(",")
+        config_file = open(file_dir, "r")
+        config_file_str = config_file.read()
+        config_file.close()
+        dictionaries_list_str = config_file_str  # .split(",")
         # <-- for splitting up multipl dictionaries in the future
         # dictionaries_list = [json.loads(d) for d in dictionaries_list_str]
-        dictionaries_list = json.loads(dictionaries_list_str)
+        dictionaries_list = ast.literal_eval(str(dictionaries_list_str))
         return dictionaries_list
 
     def fill_parameters(self, param_dicts):
-        for k in self.params.keys:
+        for k in self.params.keys():
             if k not in param_dicts:
                 param_dicts[k] = self.params[k]
         return param_dicts
 
-    def generate_model(self):
-        return IntravenousModels(self.params)
+    def generate_model(self, param_dicts):
```

Comment:
wherever you load the param dicts make it self.param_dictsto avoid passing again

Reply:
so self.param_dicts = param_dicst wherever param_dicts is defined